### PR TITLE
Add CI configuration

### PR DESCRIPTION
**Pull request by trasher**

### CODE REVIEW

File: inc/taginjection.class.php [hunk 1]
```diff
@@ -28,57 +28,66 @@
  * -------------------------------------------------------------------------
  */
 
-class PluginTagTagInjection extends PluginTagTag
-      implements PluginDatainjectionInjectionInterface {
+/** @phpstan-ignore-next-line */
```
Comment (trasher) [inline inc/taginjection.class.php:31]: Ignore does not work, error `Class PluginTagTagInjection implements unknown interface PluginDatainjectionInjectionInterface.` persists :(

File: inc/tagitem.class.php [hunk 2]
```diff
@@ -28,345 +28,371 @@
...
-   public static function uninstall() {
-      global $DB;
-
-      $migration = new Migration(PLUGIN_TAG_VERSION);
-      $migration->dropTable(getTableForItemType(__CLASS__));
-   }
+            if (method_exists($DB, 'doQueryOrDie')) {
+                $DB->doQueryOrDie($query);
+            } else {
+                /** @phpstan-ignore-next-line  */
```
Comment (trasher) [inline inc/tagitem.class.php:71]: On my local instance, there is an error:
```
 ------ ---------------------------------------------------------- 
  Line   inc/tagitem.class.php                                     
 ------ ---------------------------------------------------------- 
  71     Call to deprecated method queryOrDie() of class DBmysql:  
         10.0.11                                                   
 ------ ---------------------------------------------------------- 
```

File: inc/taginjection.class.php (hunk 1 above)
Comment (cedric-anne) [reply inc/taginjection.class.php:31]: A solution would be to add the interface in a `stubs/PluginDatainjectionInjectionInterface.php` file and add the following entry in the `phpstan.neon` file:
```yaml
    stubFiles:
        - stubs/PluginDatainjectionInjectionInterface.php
```

Comment (trasher) [reply inc/taginjection.class.php:31]: I've just tried, that does not work

Comment (trasher) [reply inc/taginjection.class.php:31]: I wonder if the best solution would not be to adddatainjection plugin

File: inc/tagitem.class.php (hunk 2 above)
Comment (trasher) [reply inc/tagitem.class.php:71]: Seems due to a local conf, but I do not find which one \o/

**Comment by cedric-anne** [comment]
You should try to add the following conf in the `phpstan.neon` file:

```yaml
    ignoreErrors:
        - '/Call to deprecated method queryOrDie() of class DBmysql/'
        - '/Class PluginTagTagInjection implements unknown interface PluginDatainjectionInjectionInterface/'
```

**Comment by trasher** [comment]
> You should try to add the following conf in the `phpstan.neon` file:
> 
> ```yaml
>     ignoreErrors:
>         - '/Call to deprecated method queryOrDie() of class DBmysql/'
>         - '/Class PluginTagTagInjection implements unknown interface PluginDatainjectionInjectionInterface/'
> ```

That does not work either:
```
Error message "Class PluginTagTagInjection implements unknown interface PluginDatainjectionInjectionInterface." cannot be ignored, use excludePaths instead.
```

File: stubs/PluginDatainjectionInjectionInterface.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ * -------------------------------------------------------------------------
+ * @copyright Copyright (C) 2007-2023 by DataInjection plugin team.
+ * @license   GPLv2 https://www.gnu.org/licenses/gpl-2.0.html
+ * @link      https://github.com/pluginsGLPI/datainjection
+ * -------------------------------------------------------------------------
+ */
+
+if (!defined('GLPI_ROOT')) {
+    die("Sorry. You can't access directly to this file");
+}
```
Comment (cedric-anne) [inline stubs/PluginDatainjectionInjectionInterface.php:33]:
```suggestion
```

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**